core/src/test/java/org/apache/iceberg/TestBase.java

```diff
@@ -173,7 +173,7 @@ public class TestBase {
   public TestTables.TestTable table = null;
 
   @Parameters(name = "formatVersion = {0}")
-  protected static List<Object[]> parameters() {
+  protected static List<Object> parameters() {
```

Comment:
why is this change necessary?

Reply:
Because for some tests, we only send 1 `Object` in the list, which is satisfied by `List<Object>`.
 For ex: from `DeleteFileIndexTestBase`
```
@Parameters(name = "formatVersion = {0}")
  public static List<Object> parameters() {
    return Arrays.asList(new Object[] {2});
  }
```

Having List<Object[]> does not allow for the above case as it always expects a List of Object arrays, where the Object arrays size cant be 1.

Reply:
in that case we should be able to change what's being returned here and in the other places that have been modified in this PR to `return Arrays.asList(1, 2);` right? That way we don't need to have the additional wrapping of params into `Object[]`

Reply:
That would work if the inputs were only Integers. But that's not the case for all child classes.
For ex: `TestSparkDistributedDataScanDeletes` inherits from `DeleteFileIndexTestBase` has additional parameters, that need to be wrapped into `Object[]`.

```
  @Parameters(name = "formatVersion = {0}, dataMode = {1}, deleteMode = {2}")
  public static List<Object> parameters() {
    return Arrays.asList(
        new Object[] {2, LOCAL, LOCAL},
        new Object[] {2, LOCAL, DISTRIBUTED},
        new Object[] {2, DISTRIBUTED, LOCAL},
        new Object[] {2, LOCAL, DISTRIBUTED});
  }
```

Reply:
yes I only meant the places that are applicable and have parameters of the same type

Reply:
If we have to use `List<Object[]>` and not `List<Object>`, we can always do this:

```
  @Parameters(name = "formatVersion = {0}")
  public static List<Object[]> parameters() {
    return Arrays.asList(new Object[][] {{2}});
  }
```

Reply:
oh just saw your previous message, got it.

Reply:
removed Object wrapping when not necessary